nano/node/vote_processor.cpp

```diff
@@ -88,56 +89,48 @@ void nano::vote_processor::process_loop ()
 	}
 }
 
-void nano::vote_processor::vote (std::shared_ptr<nano::vote> vote_a, std::shared_ptr<nano::transport::channel> channel_a)
+bool nano::vote_processor::vote (std::shared_ptr<nano::vote> vote_a, std::shared_ptr<nano::transport::channel> channel_a)
 {
+	bool process (false);
 	nano::unique_lock<std::mutex> lock (mutex);
 	if (!stopped)
 	{
-		bool process (false);
-		/* Random early delection levels
-		 Always process votes for test network (process = true)
-		 Stop processing with max 144 * 1024 votes */
-		if (!network_params.network.is_test_network ())
+		// Level 0 (< 0.1%)
+		if (votes.size () < 6.0 / 9.0 * max_votes)
 		{
-			// Level 0 (< 0.1%)
-			if (votes.size () < 96 * 1024)
-			{
-				process = true;
-			}
-			// Level 1 (0.1-1%)
-			else if (votes.size () < 112 * 1024)
-			{
-				process = (representatives_1.find (vote_a->account) != representatives_1.end ());
-			}
-			// Level 2 (1-5%)
-			else if (votes.size () < 128 * 1024)
-			{
-				process = (representatives_2.find (vote_a->account) != representatives_2.end ());
-			}
-			// Level 3 (> 5%)
-			else if (votes.size () < 144 * 1024)
-			{
-				process = (representatives_3.find (vote_a->account) != representatives_3.end ());
-			}
+			process = true;
 		}
-		else
+		// Level 1 (0.1-1%)
+		else if (votes.size () < 7.0 / 9.0 * max_votes)
 		{
-			// Process for test network
-			process = true;
+			process = (representatives_1.find (vote_a->account) != representatives_1.end ());
+		}
+		// Level 2 (1-5%)
+		else if (votes.size () < 8.0 / 9.0 * max_votes)
+		{
+			process = (representatives_2.find (vote_a->account) != representatives_2.end ());
+		}
+		// Level 3 (> 5%)
+		else if (votes.size () < max_votes)
+		{
+			process = (representatives_3.find (vote_a->account) != representatives_3.end ());
 		}
 		if (process)
 		{
 			votes.emplace_back (vote_a, channel_a);
-
-			lock.unlock ();
-			condition.notify_all ();
-			lock.lock ();
+			if (votes.size () == 1)
```

Comment:
Need some testing... All other options should be in `!votes.empty ()` `nano::vote_processor::process_loop ()` in theory

Reply:
Reverted since these changes fall outside the scope of the PR, can re-evaluate in the future.